includes/functions.php

```diff
@@ -588,6 +588,67 @@ function pmpro_getLevelsExpiration( &$levels ) {
 	return $expiration_text;
 }
 
+/**
+ * Get the text to display a membership's expiration date.
+ * @since 3.0
+ * @param int $enddate The expiration date timestamp. 
+ * @param mixed $level A level object or id (optional) to pass to filters.
+ * @param mixed $user A user object or id (optional) to pass to filters.
```

Comment:
If we want these parameters to be mixed, we should have some code to put both $level and $user into the same format before they are passed to filters. Custom code should know which format to expect data in. It would also be good to have both of these be required since we should always have a level and a user when this function is called.

If we were willing to switch both $level and $user to always be objects, we could also eliminate the $enddate parameter since we will have all the info needed to calculate $enddate from the passed $level and $user.

Reply:
I thought of this, but it introduces memory and DB overhead in cases where we're trying to avoid it (e.g. members list exports).